File: ruby/grade-school/school.rb
class School
  
  attr_reader :name, :db
  def initialize(name)
    @name = name
    @db = Hash.new { |hash, key| hash[key] = [] }
  end

  def add(student, grade)
    db[grade] << student
  end

  def grade(num)
    db[num]
  end

  def sort
    Hash[with_sorted_students.sort]
  end

  private

    def with_sorted_students
      db.map { |grade, students| [grade, students.sort] }
    end

end

File: python/grade-school/school.py
from collections import defaultdict

class School(object):

    def __init__(self, name):
        self.name = name
        self.db = defaultdict(set)

    def add(self, student, grade):
        self.db[grade].add(student)

    def grade(self, grade):
        return self.db[grade]

    def sort(self):
        return {
            grade: tuple(students)
            for grade, students in sorted(self.db.iteritems())
        }
        